### PR TITLE
Adjust _responds2records function readability

### PR DESCRIPTION
**Pull request by sregales-TNO**
While working on the #1710 I realized that the responds2records function looked like a rough draft and slightly hard to read through. So I adjust it

### CODE REVIEW

**Comment by sregales-TNO** [comment]
style ignores no longer needed with the numpy update

File: ert/storage/_storage.py [hunk 1]
```diff
@@ -396,52 +396,42 @@ def _add_numerical_data(
...
 
 
+# the local variable dataframe is not read by pylint as pandas.Dataframe,
+# but due to chunking a pandas.TextFileReader
+# https://stackoverflow.com/questions/41844485/why-the-object-which-i-read-a-csv-file-using-pandas-from-is-textfilereader-obj
 def _response2records(
     response_content: bytes, metadata: _NumericalMetaData
 ) -> ert.data.RecordCollection:
-
+    assert metadata
```
Comment (DanSava) [inline ert/storage/_storage.py:405]: Why is this needed?

File: ert/storage/_storage.py [hunk 2]
```diff
@@ -396,52 +396,42 @@ def _add_numerical_data(
...
-
-    # the local variable dataframe is not read by pylint as pandas.Dataframe,
-    # but due to chunking a pandas.TextFileReader
-    # https://stackoverflow.com/questions/41844485/why-the-object-which-i-read-a-csv-file-using-pandas-from-is-textfilereader-obj
-
-    records: List[ert.data.Record]
-    if record_type == ert.data.RecordType.LIST_FLOAT:
-        dataframe = pd.read_csv(
-            io.BytesIO(response_content), index_col=0, float_precision="round_trip"
+    if record_type not in ert.data.RecordType.__members__:
```
Comment (DanSava) [inline ert/storage/_storage.py:407]: I am not sure it is ok to use protected elements like `__members__` outside the `RecordType` class :shrug:

File: ert/storage/_storage.py [hunk 3]
```diff
@@ -396,52 +396,42 @@ def _add_numerical_data(
...
         )
-        records = [
-            ert.data.NumericalRecord(data=row.to_list())
-            for _, row in dataframe.iterrows()  # pylint: disable=no-member
-        ]
-    elif record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
-        dataframe = pd.read_csv(
-            io.BytesIO(response_content), index_col=0, float_precision="round_trip"
+
+    def interpret_data(row: pd.Series) -> Any:
```
Comment (DanSava) [inline ert/storage/_storage.py:412]: I think it would be better to pass the record type explicitly in the arguments and not rely on the `record_type` defined in the outer scope of this function.

File: ert/storage/_storage.py (hunk 1 above)
Comment (sregales-TNO) [reply ert/storage/_storage.py:405]: I don't believe it is needed but decided to keep it, with the idea that whoever set it had a reason to do so.

File: ert/storage/_storage.py (hunk 2 above)
Comment (sregales-TNO) [reply ert/storage/_storage.py:407]: `if record_type not in set(item.value for item in ert.data.RecordType)`
would that be better? or should it stay how it was?

File: ert/storage/_storage.py (hunk 3 above)
Comment (sregales-TNO) [reply ert/storage/_storage.py:412]: I had explicitly, like so:
`    def interpret_data(row: pd.Series) -> Any:
        if record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
            return {int(k): v for k, v in row.to_dict().items()} 
       if record_type == ert.data.RecordType.MAPPING_STR_FLOAT:
           return row.to_dict()
       if record_type == ert.data.RecordType.LIST_FLOAT:
           return row.to_list()
`
but pylint does not like that. it prefer what I have now or the following:
`    def interpret_data(row: pd.Series) -> Any:
        if record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
            return {int(k): v for k, v in row.to_dict().items()} 
       if record_type == ert.data.RecordType.MAPPING_STR_FLOAT:
           return row.to_dict()
      return row.to_list()
`
Should I settle for the second example for better readability?

File: ert/storage/_storage.py (hunk 1 above)
Comment (DanSava) [reply ert/storage/_storage.py:405]: As far as I can tell you added this line. There was no naked assert on the metadata before, so you are the one that needs to have a reason to do so :stuck_out_tongue_winking_eye:

File: ert/storage/_storage.py (hunk 2 above)
Comment (DanSava) [reply ert/storage/_storage.py:407]: if you want the check like then 

```python
if record_type not in set(item.value for item in ert.data.RecordType)
```
is fine by me. 

But I am not really sure this check is required at the moment, so it can be removed in my opinion

**Review by DanSava — APPROVED** [review]
Nice! :+1:
